lock-api/src/main/java/com/palantir/lock/watch/ConjureLockRequestMetadataUtils.java

```diff
@@ -0,0 +1,132 @@
+        KeyListChecksum checksum = encoded.keyListChecksum();
+        ConjureLockRequestMetadata conjureLockRequestMetadata = ConjureLockRequestMetadata.of(
+                encoded.indexToValue(), checksum.type().getId(), Bytes.from(checksum.value()));
+        return Pair.create(encoded.keyList(), conjureLockRequestMetadata);
+    }
+
+    public static LockRequestMetadata fromConjureIndexEncoded(
+            List<LockDescriptor> keyList, ConjureLockRequestMetadata conjureMetadata) {
+        ChangeMetadataFromConjureVisitor fromConjureVisitor = new ChangeMetadataFromConjureVisitor();
+        ChecksumType checksumType = ChecksumType.valueOf(conjureMetadata.getChecksumTypeId());
+        KeyListChecksum checksum = KeyListChecksum.of(
+                checksumType, conjureMetadata.getChecksumValue().asNewByteArray());
+        IndexEncodingResult<LockDescriptor, ConjureChangeMetadata> encoded =
+                IndexEncodingResult.of(keyList, conjureMetadata.getIndexToChangeMetadata(), checksum);
+        Map<LockDescriptor, ChangeMetadata> changeMetadata = IndexEncodingUtils.decode(
+                encoded, conjureChangeMetadata -> conjureChangeMetadata.accept(fromConjureVisitor));
+        // visitUnknown() will return null
+        changeMetadata.values().removeIf(Objects::isNull);
+        return LockRequestMetadata.of(changeMetadata);
+    }
+
+    private static final class ChangeMetadataToConjureVisitor implements ChangeMetadata.Visitor<ConjureChangeMetadata> {
+
+        @Override
+        public ConjureChangeMetadata visit(Unchanged unchanged) {
+            return ConjureChangeMetadata.unchanged(ConjureUnchangedChangeMetadata.of());
+        }
+
+        @Override
+        public ConjureChangeMetadata visit(Updated updated) {
+            return ConjureChangeMetadata.updated(
+                    ConjureUpdatedChangeMetadata.of(Bytes.from(updated.oldValue()), Bytes.from(updated.newValue())));
+        }
+
+        @Override
+        public ConjureChangeMetadata visit(Deleted deleted) {
+            return ConjureChangeMetadata.deleted(ConjureDeletedChangeMetadata.of(Bytes.from(deleted.oldValue())));
+        }
+
+        @Override
+        public ConjureChangeMetadata visit(Created created) {
+            return ConjureChangeMetadata.created(ConjureCreatedChangeMetadata.of(Bytes.from(created.newValue())));
+        }
+    }
+
+    private static final class ChangeMetadataFromConjureVisitor
+            implements ConjureChangeMetadata.Visitor<ChangeMetadata> {
+
+        @Override
+        public ChangeMetadata visitUnchanged(ConjureUnchangedChangeMetadata unchanged) {
+            return ChangeMetadata.unchanged();
+        }
+
+        @Override
+        public ChangeMetadata visitUpdated(ConjureUpdatedChangeMetadata updated) {
+            return ChangeMetadata.updated(
+                    updated.getOldValue().asNewByteArray(),
+                    updated.getNewValue().asNewByteArray());
```

Comment:
these conversions will always clone the underlying array

Reply:
lame, but don't see a realistic alternative, unless we make `ChangeMetadata` subclasses be ByteBuffers... which may not the worst idea? What are your thoughts?

Reply:
I am not opposed, but we should probably do this in a separate PR. I would need to do some more benchmarking to see if it's actually worth it as well. I would just leave as is in here.

Reply:
Yeah we should follow up and here and investigate how painful this would be, if at all. This could generate quite a lot of heap churn, depending on how big the values are for `ChangeMetadata`.

Reply:
(For now it's ok)